Tasks/JenkinsQueueJobV2/job.ts

```diff
@@ -92,28 +72,15 @@ export class Job {
      * This defines all and validates all state transitions.
      */
     private changeState(newState: JobState) {
-        const oldState: JobState = this.State;
-        this.State = newState;
-        if (oldState !== newState) {
-            this.debug('state changed from: ' + oldState);
-            let validStateChange: boolean = false;
-            if (oldState === JobState.New) {
-                validStateChange = (newState === JobState.Locating || newState === JobState.Streaming || newState === JobState.Joined || newState === JobState.Cut);
-            } else if (oldState === JobState.Locating) {
-                validStateChange = (newState === JobState.Streaming || newState === JobState.Joined || newState === JobState.Cut);
-            } else if (oldState === JobState.Streaming) {
-                validStateChange = (newState === JobState.Finishing);
-            } else if (oldState === JobState.Finishing) {
-                validStateChange = (newState === JobState.Downloading || newState === JobState.Queued || newState === JobState.Done);
-            } else if (oldState === JobState.Downloading) {
-                validStateChange = (newState === JobState.Done);
-            } else if (oldState === JobState.Done || oldState === JobState.Joined || oldState === JobState.Cut) {
-                validStateChange = false; // these are terminal states
-            }
-            if (!validStateChange) {
-                Util.fail('Invalid state change from: ' + oldState + ' ' + this);
-            }
+        const currnetState: JobState = this.State;
```

Comment:
LGTM. Nit: here is a little misspell.

Reply:
Thanks for the review, @egor-bryzgalov! Nice catch, I fixed it.